launcher/ui/pages/instance/StoragePage.cpp

```diff
@@ -0,0 +1,159 @@
+    m_series->setLabelsVisible();
+    m_series->setLabelsPosition(QtCharts::QPieSlice::LabelInsideHorizontal);
+    m_chart_view = new QtCharts::QChartView();
+    m_chart = new QtCharts::QChart();
+    m_chart->addSeries(m_series);
+    m_chart->setBackgroundVisible(false);
+    m_chart->setMargins(QMargins(0, 0, 0, 0));
+    m_chart->legend()->setAlignment(Qt::AlignLeft);
+    m_chart_view->setRenderHint(QPainter::Antialiasing);
+
+    ui->verticalLayout->addWidget(m_chart_view);
+    ui->retranslateUi(this);
+
+    updateCalculations();
+
+    connect(ui->button_goto_resouce_packs, &QPushButton::clicked, this, [&] { m_container->selectPage("resourcepacks"); });
+    connect(ui->button_goto_mods, &QPushButton::clicked, this, [&] { m_container->selectPage("mods"); });
+    connect(ui->button_goto_worlds, &QPushButton::clicked, this, [&] { m_container->selectPage("worlds"); });
+    connect(ui->button_goto_screenshots, &QPushButton::clicked, this, [&] { m_container->selectPage("screenshots"); });
+    connect(ui->button_goto_other_logs, &QPushButton::clicked, this, [&] { m_container->selectPage("logs"); });
+
+    connect(ui->button_clear_screenshots, &QPushButton::clicked, this, &StoragePage::handleClearScreenshotsButton);
+    connect(ui->button_clear_logs, &QPushButton::clicked, this, &StoragePage::handleClearLogsButton);
+    connect(ui->button_clear_all, &QPushButton::clicked, this, &StoragePage::handleClearAllButton);
+}
+
+StoragePage::~StoragePage()
+{
+    delete ui;
+}
+
+bool StoragePage::apply()
+{
+    return true;
+}
+
+void StoragePage::retranslate()
+{
+    ui->retranslateUi(this);
+}
+
+void StoragePage::handleClearScreenshotsButton()
+{
+    if (m_confirmation_box->exec() != QMessageBox::Yes)
+        return;
+
+    auto path = m_inst->gameRoot() + "/screenshots";
+    QDir(path).removeRecursively();
+    updateCalculations();
+}
+
+void StoragePage::handleClearLogsButton()
+{
+    if (m_confirmation_box->exec() != QMessageBox::Yes)
+        return;
+
+    auto path = m_inst->gameRoot() + "/logs";
+    QDir(path).removeRecursively();
+    updateCalculations();
+}
+
+void StoragePage::handleClearAllButton()
+{
+    if (m_confirmation_box->exec() != QMessageBox::Yes)
+        return;
+
+    handleClearScreenshotsButton();
+    handleClearLogsButton();
+}
+
+void StoragePage::updateCalculations()
+{
+    m_size_resource_packs =
+        getDirectorySize((m_inst->gameRoot() + "/texturepacks")) + getDirectorySize((m_inst->gameRoot() + "/resourcepacks"));
+    m_size_mods = getDirectorySize(m_inst->modsRoot());
+    m_size_saves = getDirectorySize((m_inst->gameRoot() + "/saves"));
+    m_size_screenshots = getDirectorySize((m_inst->gameRoot() + "/screenshots"));
+    m_size_logs = getDirectorySize((m_inst->gameRoot() + "/logs"));
+
+    auto storage_info = QStorageInfo(QDir(m_inst->gameRoot()));
+    m_size_remaining = storage_info.bytesAvailable();
+    m_size_used = storage_info.bytesTotal() - m_size_remaining;
```

Comment:
https://doc.qt.io/qt-6/qstorageinfo.html#QStorageInfo-1
> If you pass a directory or file, the QStorageInfo object will refer to the volume where this directory or file is located. You can check if the created object is correct using the [isValid](https://doc.qt.io/qt-6/qstorageinfo.html#isValid)() method.

In other words, this gets the used/total size of whichever storage device the instance directory happens to be on. Is this intentional? I thought it should get the used space of just the instance directory, and the remaining space of the storage medium which it is on, and report those figures to the user.

Reply:
Yes, this is intentional.